lib/matplotlib/markers.py

```diff
@@ -677,7 +677,7 @@ def _set_pentagon(self):
             self._path = polypath
         else:
             verts = polypath.vertices
-            y = (1 + np.sqrt(5)) / 4.
+            y = (1 + 5 ** (1 / 2)) / 4.
```

Comment:
I'm not convinced of all these sqrt micro-optimizations. IMHO it makes the code less readable, and the performance gain is minimal. For example here

The sqrt is less than 2% of the the time of the immediate surrounding code in the function:

``` 
In [21]: %%timeit
    ...: Affine2D().scale(0.5)
    ...: polypath = Path.unit_regular_polygon(5)
    ...: verts = polypath.vertices
    ...: top = Path(verts[[0, 1, 4, 0]])
    ...: bottom = Path(verts[[1, 2, 3, 4, 1]])
    ...: left = Path([verts[0], verts[1], verts[2], [0, -y], verts[0]])
    ...: right = Path([verts[0], verts[4], verts[3], [0, -y], verts[0]])
    ...: 
61.1 µs ± 228 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [22]: %timeit np.sqrt(5)
2.36 µs ± 35 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

and that function `_set_pentagon()` again is only used to create a new marker object, which in itself is quite rare. So I bet the performance gain is not measurable in any real example.